### PR TITLE
[Placement Group] Capture child tasks by default.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR does 2 things.
- Turn on the implicit pg capture by default.
- Support an option to turn off this option.

To enable the second option, this is how it works.

- If the capture child task is specified, the receiver configures this context to be true.
- When the receiver submits a new task, we check the worker context and decide whether or not if we want to configure this option.

Missing piece: Documentation (will create another PR since it is already pretty big).

Open question: Adding a new option makes the PR like 10 times bigger. Should we just not consider this option? 

## Related issue number

https://github.com/ray-project/ray/issues/9808

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/context.cc [hunk 1]
```diff
@@ -90,6 +99,10 @@ struct WorkerThreadContext {
   ///   thread local placement group id for tasks, and the process placement group id for
   ///   actors.
   PlacementGroupID current_placement_group_id_;
+
+  /// Whether or not if the current task should capture the parent's placement group
```
Comment (ericl) [inline src/ray/core_worker/context.cc:103]:
```suggestion
  /// Whether or not child tasks are captured in the parent's placement group
```

File: src/ray/core_worker/context.cc [hunk 2]
```diff
@@ -90,6 +99,10 @@ struct WorkerThreadContext {
...
+  /// implicilty.
```
Comment (ericl) [inline src/ray/core_worker/context.cc:104]:
```suggestion
  /// implicitly.
```

File: src/ray/core_worker/context.cc [hunk 3]
```diff
@@ -134,6 +148,15 @@ const PlacementGroupID &WorkerContext::GetCurrentPlacementGroupId() const {
   }
 }
 
+bool WorkerContext::PlacementGroupCaptureChildTasks() const {
```
Comment (ericl) [inline src/ray/core_worker/context.cc:151]:
```suggestion
bool WorkerContext::ShouldCaptureChildTasksInPlacementGroup() const {
```

File: src/ray/core_worker/core_worker.h [hunk 4]
```diff
@@ -358,6 +358,10 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
     return worker_context_.GetCurrentPlacementGroupId();
   }
 
+  bool ShouldCaptureParentPlacementGroup() const {
```
Comment (ericl) [inline src/ray/core_worker/core_worker.h:361]:
```suggestion
  bool ShouldCaptureChildTasksInPlacementGroup() const {
```

**Review by ericl — CHANGES_REQUESTED** [review]
Mostly naming comments

**Review by ericl — APPROVED** [review]

File: src/ray/core_worker/context.cc [hunk 5]
```diff
@@ -134,6 +147,15 @@ const PlacementGroupID &WorkerContext::GetCurrentPlacementGroupId() const {
   }
 }
 
+bool WorkerContext::ShouldCaptureChildTasksInPlacementGroup() const {
+  // If the worker is an actor, we should return the actor's placement group id.
```
Comment (ffbin) [inline src/ray/core_worker/context.cc:151]: We do not return placement group id?

**Comment by ffbin** [comment]
@rkooo567 Does Java API need to be modified?

**Comment by rkooo567** [comment]
@ffbin Yes. There are some parity to Java APIs from Python APIs, and we should unify them. I believe we also don't have ready API.

File: src/ray/core_worker/context.cc (hunk 5 above)
Comment (rkooo567) [reply src/ray/core_worker/context.cc:151]: ah, good catch. Let me fix.

**Review by ffbin — APPROVED** [review]

**Pull request merged**